artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/federation/internal/FederationQueueEntry.java

```diff
@@ -28,19 +33,19 @@
  * on a federation resource such that it is not torn down until all demand has been
  * removed from the local resource.
  */
-public class FederationConsumerEntry {
+public class FederationQueueEntry {
 
    private final FederationConsumerInternal consumer;
 
-   private int references = 1;
+   private final Set<String> consumerDemand = new HashSet<>();
```

Comment:
I was wondering why you didn't just use a HashSet of ServerConsumers here.

But I guess this is better since it decreases chances of memory leaking? if you forgot an entry here at least you wouldn't leak the consumer->connection and everything associated with it?

Reply:
I wanted to avoid any chance of leaking the ServerConsumer instance which have lead to holding onto other resources in the past.  If the notification isn't sent and we don't remove the demand at least we aren't holding onto the actual consumer.

Reply:
@tabish121 that's what I thought why you did it.. thanks